src/main/telemetry/smartport.c

```diff
@@ -538,7 +538,11 @@ void handleSmartPortTelemetry(void)
             }
         }
 
-        if (cmpTimeUs(micros(), lastTelemetryFrameReceivedUs) >= SMARTPORT_MIN_TELEMETRY_RESPONSE_DELAY_US) {
+        // XXX: Checking for lastTelemetryFrameReceivedUs == 0 is needed for S.Port, since payload will
+        // only be non-null (and hence, lastTelemetryFrameReceivedUs will be updated) when using FPort.
```

Comment:
On closer inspection, this is actually wrong. Payload is pointing to the incoming payload for both SmartPort and FPort:
https://github.com/iNavFlight/inav/pull/3536/files#diff-5d6c27b6b68ea6ef24a6103f1bc2ced4L535

A correct fix would be to change

https://github.com/iNavFlight/inav/pull/3536/files#diff-5d6c27b6b68ea6ef24a6103f1bc2ced4R536

to 

```
if (payload || clearToSend) {
```

Also, n.b. that `handleSmartPortTelemetry()` isn't run for FPort.

Reply:
I thought about that, but clearToSend might reset lastTelemetryUs if we get an stray byte due to intererence. I wanted to fix it with minimal changes so it can go into inav 2.0